post-meta-inspector.php

```diff
@@ -111,13 +110,62 @@ public function post_meta_inspector() {
 				word-wrap: break-word;
 			}
 		</style>
+		<?php self::render_table(); ?>
+		<script>
+			jQuery(document).ready(function() {
+				jQuery('.pmi_toggle').click( function(e){
+					jQuery('+ code', this).show();
+					jQuery(this).hide();
+				});
+
+				<?php if ( function_exists( 'is_gutenberg_page' ) && is_gutenberg_page() ) : ?>
+				var editPost = wp.data.select( 'core/edit-post' ), lastIsSaving = false;
+
+				wp.data.subscribe( function() {
+					var isSaving = editPost.isSavingMetaBoxes();
+					if ( isSaving !== lastIsSaving && ! isSaving ) {
+						lastIsSaving = isSaving;
+						// Gutenberg Post Saving has finished!
+						var data = {
+							'action': 'update_post_meta_inspector',
+							'nonce': <?php echo wp_json_encode( wp_create_nonce( 'update_post_meta_inspector' ) ); ?>,
+							'post': <?php echo wp_json_encode( $post_id ); ?>
+						};
+
+						jQuery.get( ajaxurl, data, function( response ) {
+							jQuery( '#post_meta_inspector' ).html( response );
+						} );
+					}
+
+					lastIsSaving = isSaving;
+				} );
+				<?php endif; ?>
+			});
+		</script>
+		<?php
+	}
 
-		<?php $custom_fields = get_post_meta( get_the_ID() ); ?>
-		<table>
+	/**
+	 * Renders the Post Meta table.
+	 *
+	 * @return void
+	 */
+	public function render_table() {
+		$post_id           = isset( $_GET['post'] ) ? (int) $_GET['post'] : get_the_ID();
+		$custom_fields     = get_post_meta( $post_id );
+		$toggle_length     = apply_filters( 'pmi_toggle_long_value_length', 0 );
+		$toggle_length     = max( intval( $toggle_length ), 0 );
+		$toggle_el_escaped = '<a href="javascript:void(0);" class="pmi_toggle">' . esc_html__( 'Click to show&hellip;', 'post-meta-inspector' ) . '</a>';
+
+		if ( wp_doing_ajax() ) {
```

Comment:
Rather than multiple `if` clauses and flows within this function, I would add a separate AJAX handler which handles the nonce and capability check (which is missing, btw) and calls `render_table`.